### DISCUSSION TITLE
Plane with finite support/extent

### DISCUSSION THREAD

**Discussion by jeremyong** [jrouwe/JoltPhysics#1281] [General]
I noticed that creating a box with 0 extent along one axis (a square essentially) will trip an assert due to the convex radius exceeding the minimum box extent. I suppose one way to accomplish this would be to make a compound shape with 4 triangles (2 triangles per square, 2 squares with opposing orientation). Is there a more suitable shape that describes this use case?

I'm mentioning this also because the documentation seemed to indicate that a convex radius of 0 was permitted but the asserts in various shape types seem to indicate otherwise. Thanks!

**Comment by jrouwe** [comment]
What assert do you get?

I tried it and I only get a division by zero in `DebugRenderer::DrawBox` which I fixed in https://github.com/jrouwe/JoltPhysics/commit/81c98592a50e4c8d74415175ec7091c537045377, but I doubt that's what you're seeing.

B.t.w. using very thin shapes is not a great idea in general. It is very easy for objects to pass through them.

**Reply by jeremyong** [reply]
Sorry I should have included that in the initial question:

```
	if (inSettings.mConvexRadius < 0.0f
		|| inSettings.mHalfExtent.ReduceMin() <= inSettings.mConvexRadius)
	{
		outResult.SetError("Invalid convex radius");
		return;
	}
```

The issue here being that the convex radius must be greater than the minimum extent and non-zero.

**Reply by jrouwe** [reply]
Setting the convex radius to 0 is required when you have a box with edge length 0 (and is also what I did while testing). What asserts are you getting with convex radius 0?

**Reply by jeremyong** [reply]
Ah that's embarrassing. I was still getting the "Invalid convex radius" assert but it turns out I had a bug in the routine that tried to infer the convex radius from the shape bounds (including the zero extent handling). I will close the issue and keep your comment about thin shape issues in mind. Thanks!
